### ISSUE TITLE
Paypal Express fails for virtual products - Magento 2.1.8 CE

### ISSUE THREAD

**Issue by darryldavidson** [magento/magento2#10615]
Checkout using Paypal Express fails when we have virtual product in the cart. The page redirects to the review order page, with "Can't place order" message. When we then click "place order" from the review page the order is placed fine.

### Preconditions
1.  Magento 2.1.8 CE without sample data, CentOS 7, php 7, NGINX, MariaDB 10.2.6
2.  Paypal Express setup in sandbox mode - in context method.
3.  Paypal advanced setting "Skip review page" set to Yes
4.  Single test category added
5.  Single virtual product added and assigned to test category
6.  No other changes made from default 2.1.8 version 

### Steps to reproduce
1.  Navigate to test category
2.  Add virtual product to basket
3.  Proceed to checkout
4.  Choose "Paypal Express" - this is the only payment method added
5.  In context paypal window pops-up, enter login detail for customer sandbox
6.  Click 'Pay now' in Paypal window

### Expected result
1.  The success page should be displayed after step 6 above

### Actual result
1.  Returns to website review page with error message "We can't place the order."
2.  Click "Place Order" - completes fine and displays success page

3.  Error in var/log/system.log
```
main.CRITICAL: Exception message: Please check the billing address information. Please enter the street. Please enter the city. Please enter the phone number.
Trace: #0 /var/www/samba/gsc-magento/vendor/magento/module-quote/Model/QuoteManagement.php(427): Magento\Quote\Model\QuoteValidator->validateBeforeSubmit(Object(Magento\Quote\Model\Quote))
#1 /var/www/samba/gsc-magento/vendor/magento/module-quote/Model/QuoteManagement.php(391): Magento\Quote\Model\QuoteManagement->submitQuote(Object(Magento\Quote\Model\Quote), Array)
#2 /var/www/samba/gsc-magento/vendor/magento/module-quote/Model/QuoteManagement.php(349): Magento\Quote\Model\QuoteManagement->submit(Object(Magento\Quote\Model\Quote))
```

The issue seems to come from a change in version 2.1.8 in file vendor\magento\module-paypal\Model\Express\Checkout.php at line 808 

```
$orderId = $this->quoteManagement->placeOrder($this->_quote->getId());
$order = $this->getOrderRepository()->get($orderId);
```

We have worked around the issue for now by rolling back these lines as per 2.1.7

```
//        $orderId = $this->quoteManagement->placeOrder($this->_quote->getId());
//        $order = $this->getOrderRepository()->get($orderId);
        $order = $this->quoteManagement->submit($this->_quote);
```

**Comment by YevSent** [comment]
Hi, @darryldavidson, thanks for reporting. I've created the internal ticket (MAGETWO-71821) for a future investigation.

**UPD**: already fixed on the develop branch https://github.com/magento/magento2/commit/0b30eea404328c630947c0ba4c6aa75f69ae469d#diff-caa0bd0cad3defa70013c99c510f64c9.

**Comment by magento-engcom-team** [comment]
Hi @darryldavidson 
Thank you for report. The issue is fixed in c680f1ca0965b475ee3146fc75ea231e020bf0cf and will be avalible in 2.1.10 release soon
